routes/ci.js

```diff
@@ -0,0 +1,125 @@
+var express = require("express");
+var router = express.Router();
+var request = require("request")
+var repositoryModel = require("../model/repository.js");
+var crypter = require("../util/crypter.js");
+var generator = require("../backend/generator.js");
+var deploy = require("../backend/deploy.js");
+router.get('/register', function (req, res, next) {
+  request({
+    url: `https://api.github.com/users/${req.session.username}/repos?per_page=100`,
+    headers: {
+      'User-Agent': 'request'
+    }
+  },
+  function (error, response, body) {
+    var repos = JSON.parse(body);
+    var query = repos.map(x => `repos=${x.name}`);
+    query = query.join("&");
+    res.redirect(`/?${query}`)
+  })
+})
+
+router.post('/register', function (req, res, next) {
+
+  var repositoryName = req.body.repository;
+  request({
+    url: `https://api.github.com/repos/${req.session.username}/${repositoryName}/hooks?access_token=${req.session.token}`,
+    headers: {
+      'User-Agent': 'request'
+    }
+  }, function (error, response, body) {
+    var hooks = JSON.parse(body);
+    var hookurl = process.env.HOSTNAME + '/ci/webhook'
+    var isRegistered = false;
+    hooks.forEach(function (hook) {
+      if (hook.config.url == hookurl) {
+        isRegistered = true
+      }
+    })
+    if (!isRegistered) {
+      request({
+        url: `https://api.github.com/repos/${req.session.username}/${repositoryName}/hooks?access_token=${req.session.token}`,
+        headers: {
+          'User-Agent': 'request'
+        },
+        method: 'POST',
+        json: {
+          name: "web",
+          active: true,
+          events: [
+            "push"
+          ],
+          config: {
+            url: hookurl,
+            content_type: "json"
+          }
+        }
+      }, function(error, response, body) {
+        repositoryModel.newRepository(repositoryName,
+          req.session.username,
+          req.session.githubId,
+          crypter.encrypt(req.session.token),
+          req.session.email)
+          .then(function(result) {
+            res.redirect("/?message=Thanks for registering with Yaydoc.Hereafter Documentation will be pushed to the GitHub pages on each commit.");
```

Comment:
instead of spitting out the message as a get request, use `/?status=true` or something like '/?status=200' 
and in `index.jade` line 119 check for this status and display the message